Assets/Mirror/Runtime/NetworkWriter.cs

```diff
@@ -19,28 +19,50 @@ public class NetworkWriter
 
         // 'int' is the best type for .Position. 'short' is too small if we send >32kb which would result in negative .Position
         // -> converting long to int is fine until 2GB of data (MAX_INT), so we don't have to worry about overflows here
-        public int Position;
-
+        int position;
         int length;
 
-        public int Length
+        public int Position
         {
-            get => length;
-            private set
+            get => position;
+            set
             {
+                position = value;
                 EnsureCapacity(value);
-                length = value;
-                if (Position > length)
-                    Position = length;
             }
         }
 
+        /// <summary>
+        /// Reset both the position and length of the stream
+        /// </summary>
+        /// <remarks>
+        /// Leaves the capacity the same so that we can reuse this writer without extra allocations
+        /// </remarks>
+        public void Reset()
+        {
+            position = 0;
+            length = 0;
+        }
+
+        [System.Obsolete("Use Reset instead", true)]
+        public void SetLength(int value)
+        {
+            position = 0;
+            length = 0;
+        }
+
+
         void EnsureCapacity(int value)
```

Comment:
Very misleading name,  EnsureCapacity sounds like making sure we have enough capacity, which I would not expect to modify the length.

A new name would make what is going on here clearer

```suggestion
        void EnsureLength(int value)
```